### ISSUE TITLE
💸 GitConsensus addRelease(): use transfer() instead of mint() for distributions

### ISSUE THREAD

**Issue by mattstam** [git-consensus/contracts#29]
Instead of [token.mint()](https://github.com/git-consensus/contracts/blob/842a3e33188010ed2de8506acb71d7f34351839a/contracts/GitConsensus.sol#L101), we could use the standard ERC20 [token.transfer()](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/561d0eead30d3977847f2c2b4ae54eac31c218df/contracts/token/ERC20/IERC20.sol#L41) if the GitConsensus contract is an approved to transfer that amount before-hand.

This would create alternative tokenomics models, such as having as designating a repository treasury. This type of model would could allow for a stable `tokenSupply()`, which is particularly desirable for those projects looking to maintain a stable trading price overtime.

**Comment by DmitriyShepelev** [comment]
Having the ability to `mint` and `transfer` would be a good idea. For example, if Alice thinks Bob made a particularly impactful commit to a repo R but doesn't want to increase the supply of the token for R (and thereby diluting the value of her tokens), she can transfer a subset of her tokens.

Having a dedicated repository treasury is an interesting idea. Project owners can have their repository treasury mint X tokens every Y blocks to ensure supply stability.

**Comment by mattstam** [comment]
> For example, if Alice thinks Bob made a particularly impactful commit to a repo R but doesn't want to increase the supply of the token for R (and thereby diluting the value of her tokens), she can transfer a subset of her tokens.

Yep! Trying to play devil's advocate here, and this goes through my mind:
1. If Alice wants to reward Bob, she can just `transfer()` her R tokens to Bob directly instead of having to go through GitConsensus and with correlated commit(s).
2. @Bryce-Soghigian mentioned that status tracking as a particularly useful (e.g. if you made a contribution to a major project and got minted tokens, it is worth showing in a career-oriented page / resume). With minting this is simple to track because it's just tracking `transfer(0x0, bobAddr)`. With transfer rewards, the first `from` address can be arbitrary so harder to filter.

> Project owners can have their repository treasury mint X tokens every Y blocks to ensure supply stability.

What mechanisms currently exist for periodic calls like this? In my experience there needs to be something external (usually an EOA) to trigger any computation / state changes. I've seen periodic contract pollers a while ago, but am really behind on what the current patterns for this would be.

**Comment by DmitriyShepelev** [comment]
> With transfer rewards, the first `from` address can be arbitrary so harder to filter.
Could we emit an event utilizing a minimal amount of gas to identify transfer rewards?
> 
> > Project owners can have their repository treasury mint X tokens every Y blocks to ensure supply stability.
> 
> What mechanisms currently exist for periodic calls like this? In my experience there needs to be something external (usually an EOA) to trigger any computation / state changes. I've seen periodic contract pollers a while ago, but am really behind on what the current patterns for this would be.

Two ways come to mind:
1. Use Chainlink Keepers ([docs](https://docs.chain.link/docs/chainlink-automation/compatible-contracts/)) to create a `checkUpkeep` function to see if a certain number of blocks have been mined.
2. Use [a modifier](https://docs.soliditylang.org/en/v0.8.13/contracts.html?highlight=modifier#function-modifiers) that runs after each function checking if a certain number of blocks have been mined, the block number is divisible by some sort of number, etc.

(2) would probably be more gas-efficient.
